File: src/Shared/Dtos/DuplicateGroupDto.cs
namespace Shared.Dtos;

/// <summary>
/// Data transfer object for duplicate file groups.
/// </summary>
public record DuplicateGroupDto
{
    public Guid Id { get; init; }
    public required string Hash { get; init; }
    public int FileCount { get; init; }
    public long TotalSize { get; init; }
    public long PotentialSavings => FileCount > 1 ? TotalSize - (TotalSize / FileCount) : 0;
    public DateTime? ResolvedAt { get; init; }
    public DateTime CreatedAt { get; init; }
    public Guid? OriginalFileId { get; init; }
    public IReadOnlyList<IndexedFileDto> Files { get; init; } = [];

    /// <summary>
    /// Thumbnail path of the first file in the group (for list display).
    /// </summary>
    public string? FirstFileThumbnailPath { get; init; }

    // Validation fields
    public string Status { get; init; } = "pending";
    public DateTime? ValidatedAt { get; init; }
    public Guid? KeptFileId { get; init; }

    // Review session fields
    public DateTime? LastReviewedAt { get; init; }
    public int? ReviewOrder { get; init; }
    public Guid? ReviewSessionId { get; init; }
}

File: src/Web/src/app/models/duplicate-group.dto.ts
import { IndexedFileDto } from './indexed-file.dto';

/**
 * Data transfer object for duplicate file groups.
 * Matches backend: src/Shared/Dtos/DuplicateGroupDto.cs
 */
export interface DuplicateGroupDto {
  id: string;
  hash: string;
  fileCount: number;
  totalSize: number;
  potentialSavings: number;
  resolvedAt: string | null;
  createdAt: string;
  originalFileId: string | null;
  files: IndexedFileDto[];

  // Thumbnail for list display
  firstFileThumbnailPath: string | null;

  // Validation fields
  status: string;
  validatedAt: string | null;
  keptFileId: string | null;

  // Review session fields
  lastReviewedAt: string | null;
  reviewOrder: number | null;
  reviewSessionId: string | null;
}

/**
 * Request to set a file as the original in a duplicate group.
 * Matches backend: src/Shared/Requests/SetOriginalRequest.cs
 */
export interface SetOriginalRequest {
  fileId: string;
}
